pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -262,8 +296,146 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
 	return preds
 }
 
-func (s *cyclonedxParser) getPackageElement(elementID string) []*model.PkgInputSpec {
-	if packNode, ok := s.packagePackages[string(elementID)]; ok {
+func (c *cyclonedxParser) getVulnerabilities(ctx context.Context, cdxBom *cdx.BOM) *vulnData {
+	if cdxBom.Vulnerabilities == nil {
+		return nil
+	}
+
+	var vInfo vulnData
+	var status model.VexStatus
+	var justification model.VexJustification
+	var publishedTime time.Time
+	layout := "2006-01-02T15:04:05.000Z"
+	for _, vulnerability := range *c.cdxBom.Vulnerabilities {
+		vuln, err := asmhelpers.CreateVulnInput(vulnerability.ID)
+		if err != nil {
+			return nil
+		}
+
+		if vexStatus, ok := vexStatusMap[vulnerability.Analysis.State]; ok {
+			status = vexStatus
+		}
+
+		if vexJustification, ok := justificationsMap[vulnerability.Analysis.Justification]; ok {
+			justification = vexJustification
+		}
+
+		time, err := time.Parse(layout, vulnerability.Published)
+		if err == nil {
+			publishedTime = time
+		}
+
+		vd := model.VexStatementInputSpec{
+			Status:           status,
+			VexJustification: justification,
+			KnownSince:       publishedTime,
+			Statement:        vulnerability.Analysis.Detail,
+			StatusNotes:      fmt.Sprintf("%s:%s", string(vulnerability.Analysis.State), string(vulnerability.Analysis.Justification)),
+		}
+
+		for _, affect := range *vulnerability.Affects {
+			vi := c.getAffectedPackages(ctx, vuln, vd, affect)
+			if vi == nil {
+				continue
```

Comment:
log here and continue

Reply:
any specific values we want to log here given that vi equates to nil?

Reply:
again move all this to the parse function